libraries/AP_Scripting/lua_boxed_numerics.cpp

```diff
@@ -19,7 +19,9 @@ uint32_t coerce_to_uint32_t(lua_State *L, int arg) {
 
         int success;
         const lua_Integer v = lua_tointegerx(L, arg, &success);
-        if (success && v >= 0) {
+        // lua_tointegerx() will succeed for both signed and unsigned integers
```

Comment:
needs discussion with @WickedShell

Reply:
ping @WickedShell

Reply:
help @WickedShell

Reply:
I'm having trouble really selling myself on this. I get why it's desired in the change set, but I also think it opens some oddities as we now sometimes accept negative numbers, sometimes we won't (such as float). Given what the FrSky stuff wants can we just pass in a `int32_t` instead of `uint32_t` ? Is there a value to the uint32_t type here that I've missed?

Reply:
@WickedShell I need the full uint32_t range, I pack numbers with the AP_Frsky_SPort::prep_number() so 0xffffffff is a potential valid value, int32_t simply isn't large enough

Reply:
...but I really care about the bits not about the value, so I might try passing an int32 with a negative value but with the correct bits